File: ts/examples/simple-example-ts/src/events/userCreatedRuntimeEvent.ts
import {Description, DisplayMessage, Identifier, Name, Signature} from "@code0-tech/hercules";

@Identifier("user_created")
@Signature("(): {userId: number}")
@Name({code: "en-US", content: "User created event"})
@DisplayMessage({code: "en-US", content: "Triggers on user creation"})
@Description({code: "en-US", content: "Triggers on user creation and has a payload including the user database id"})
export class UserCreatedRuntimeEvent {}

File: py/examples/simple-example-py/events/user_created_runtime_event.py
from hercules import Description, DisplayMessage, Identifier, Name, Signature


@Identifier("user_created")
@Signature("(): {userId: number}")
@Name({"code": "en-US", "content": "User created event"})
@DisplayMessage({"code": "en-US", "content": "Triggers on user creation"})
@Description(
    {
        "code": "en-US",
        "content": "Triggers on user creation and has a payload including the user database id",
    }
)
class UserCreatedRuntimeEvent:
    pass
